.github/workflows/reusable-workflow-e2e.yaml

```diff
@@ -0,0 +1,397 @@
+      with:
+        repository: openstack-k8s-operators/openstack-operator
+
+    - name: Edit custom-bundle.Dockerfile
+      run: |
+        sed -i "s/quay.io\/openstack-k8s-operators\/${{ inputs.operator_name }}-bundle:latest/${{ env.imageregistry }}\/${{ env.imagenamespace }}\/${{ inputs.operator_name }}-bundle:${{ github.sha }}/" ${{ github.workspace }}/custom-bundle.Dockerfile
+
+    - name: Install operator-sdk
+      uses: redhat-actions/openshift-tools-installer@v1
+      with:
+        source: github
+        operator-sdk: '1.23.0'
+
+    - name: Create bundle image
+      run: |
+        pushd ${{ github.workspace }}/.github/
+        chmod +x "create_bundle.sh"
+        "./create_bundle.sh"
+        popd
+      env:
+        REGISTRY:  ${{ env.imageregistry }}/${{ env.imagenamespace }}
+        GIT_SHA: ${{ github.sha }}
+        BASE_IMAGE: openstack-operator
+
+    - name: Build openstack-operator-bundle using buildah
+      id: build-openstack-operator-bundle
+      uses: redhat-actions/buildah-build@v2
+      with:
+        image: openstack-operator-bundle
+        tags: ${{ github.sha }}
+        containerfiles: |
+          ./custom-bundle.Dockerfile
+
+    - name: Push openstack-operator To ${{ env.imageregistry }}
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: ${{ steps.build-openstack-operator-bundle.outputs.image }}
+        tags: ${{ steps.build-openstack-operator-bundle.outputs.tags }}
+        registry:  ${{ env.imageregistry }}/${{ env.imagenamespace }}
+        username: ${{ secrets.RDO_QUAY_USERNAME }}
+        password: ${{ secrets.RDO_QUAY_PASSWORD }}
+
+
+  build-openstack-operator-index:
+    needs: [build-openstack-operator-bundle]
+    name: Build openstack-operator-index Image
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: Checkout openstack-operator repository
+      uses: actions/checkout@v2
+      with:
+        repository: openstack-k8s-operators/openstack-operator
+
+    - name: Install opm
+      uses: redhat-actions/openshift-tools-installer@v1
+      with:
+        source: github
+        opm: 'latest'
+
+    - name: Create index image
+      run: |
+        pushd ${{ github.workspace }}/.github/
+        chmod +x "create_opm_index.sh"
+        "./create_opm_index.sh"
+        popd
+      env:
+        REGISTRY:  ${{ env.imageregistry }}/${{ env.imagenamespace }}
+        GIT_SHA: ${{ github.sha }}
+        BUNDLE_IMAGE: openstack-operator-bundle
+        INDEX_IMAGE_TAG: ${{ github.sha }}
+        INDEX_IMAGE: openstack-operator-index
+
+    - name: Push openstack-operator-index To ${{ env.imageregistry }}
+      uses: redhat-actions/push-to-registry@v2
+      with:
+        image: openstack-operator-index
+        tags: ${{ github.sha }}
+        registry:  ${{ env.imageregistry }}/${{ env.imagenamespace }}
+        username: ${{ secrets.RDO_QUAY_USERNAME }}
+        password: ${{ secrets.RDO_QUAY_PASSWORD }}
+
+
+  SNO-e2e-install-operator:
```

Comment:
I would consider breaking in 2 workflows: 1) building images/bundles/catalogs and 2) installing and testing. A job that only test image builds can be added for all operators without worring about resource limitations with OCP deploy.
If that make sense, we can push a new PR for that.

Reply:
Thanks Doug,

When jobs are added in the same workflow, we can create dependency chains which allow us to mention that second job will only trigger after first job. If we break these jobs in different workflows, creating dependency that SNO job will trigger after operator image build will be difficult.

Currently, We plan to add SNO e2e job on each operator, but if there is a need to only have operator image build job then we can i.e write a new workflow for that / Add some condition on SNO job that it will skip on certain condition/ Use nested reusable workflow(break in two workflow). But lets consider that later based on CI jobs requirement.